### PR TITLE
Added Industrial Miner & Removed Digital Miner

### PR DESCRIPTION
**Pull request by TheBusyBiscuit**
## Description

The Digital Miner has been deprecated for a very long time, now it's finally been removed.

This Pull Request proposes a new machine that functions similar but is much more performance-friendly. It is also much more balanced as it consumes fuel to run.
It features beautiful animations and particle effects.
When operating this machine, there will be a short warm up animation with the Pistons moving up and down.

Here's a video showcase:
https://youtu.be/1hvv6hHNCiI

## Changes

* Removed Digital Miner
* Added Industrial Miner
* Added Advanced Industrial Miner
* Also added Dried Kelp blocks to the Coal Generator

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [x] I did my best to add documentation to any public classes or methods I added.
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

**Comment by TheBusyBiscuit** [comment]
@WalshyDev I summon thou for you shall review.

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/miner/ActiveMiner.java [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+
+    private int fuel = 0;
+    private int ores = 0;
+    private boolean running = false;
+
+    private final Block chest;
+    private final Block[] pistons;
+
+    private final Location start;
+    private final Location end;
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/miner/ActiveMiner.java:51]: There's BlockPosition in CS-CoreLib2 which would be better for this.

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/miner/ActiveMiner.java [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+                    }
+                }
+
+                nextColumn();
+            });
+
+            queue.execute(SlimefunPlugin.instance);
+        }
+        catch (Exception e) {
+            Slimefun.getLogger().log(Level.SEVERE, "An Error occured while running an Industrial Miner", e);
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/miner/ActiveMiner.java:202]: We should post location info like in cargo. Better yet, have a util method in ErrorCatcher to handle it for us.

Think we should always post coords if possible, server owners may need them depending on the issue.

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/miner/ActiveMiner.java [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+                    stop("machines.INDUSTRIAL_MINER.piston-space");
+                }
+            }
+            else {
+                // The piston has been destroyed
+                stop("machines.INDUSTRIAL_MINER.destroyed");
+            }
+        }
+        catch (Exception e) {
+            Slimefun.getLogger().log(Level.SEVERE, "An Error occured while moving a Piston for an Industrial Miner", e);
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/miner/ActiveMiner.java:357]: same as said above

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/miner/IndustrialMiner.java [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+ * @author TheBusyBiscuit
+ * 
+ * @see AdvancedIndustrialMiner
+ * @see ActiveMiner
+ *
+ */
+public class IndustrialMiner extends MultiBlockMachine {
+
+    protected final Map<Location, ActiveMiner> activeMiners = new HashMap<>();
+    protected final List<MachineFuel> fuelTypes = new ArrayList<>();
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/miner/IndustrialMiner.java:46]: does this need to be a list? If you need order a LinkedHashSet may be better.

Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/miner/IndustrialMiner.java:46]: It does need to be consistently ordered, don't really see the benefits of using a LinkedHashSet though here, an ArrayList is just fine for this purpose.

**Pull request merged**